core/src/main/java/gov/nist/secauto/metaschema/core/model/constraint/DefaultConstraintValidator.java

```diff
@@ -676,10 +676,16 @@ public void finalizeValidation(DynamicContext dynamicContext) {
           try {
             List<String> key = IIndex.toKey(item, constraint.getKeyFields(), dynamicContext);
 
-            INodeItem referencedItem = index.get(key);
+            if (index == null) {
+              getConstraintValidationHandler().handleGenericValidationViolation(constraint, keyRef.getNode(), item,
+                  "Key reference to undefined index");
```

Comment:
Can you include the name of the index in the message?

Reply:
Added in https://github.com/usnistgov/metaschema-java/pull/282/commits/b52adce288be1859bc10e4523ffd42f146a82875.